### ISSUE TITLE
Debian guest allways say Login Incorrect

### ISSUE THREAD

**Issue by dianaxxyyzz** [lxc/lxc#1339]
Hello , I run debian inside Android host using lxc.
I run lxc-start -n jessie , and at login prompt no matter I enter the good password ,it say log in incorrect.
Also I did lxc-attact -n jessie ,I exported PATH , all is good inside debian , network works etc .

But when I try to change a password it say : 

root@jessie:/# passwd
Enter new UNIX password: 
Retype new UNIX password: 
passwd: System error
passwd: password unchanged
root@jessie:/# 

What could be the problem?

**Comment by brauner** [comment]
`Debian Jessie` containers are not working correctly in a containerized environment since the `systemd` version it ships is too old. So I suspect this is a side effect of that problem.

**Comment by stgraber** [comment]
Yeah, I suspect the system error comes from some pam-systemd integration failing.

**Comment by brauner** [comment]
Closing, since there's nothing that we can do about this atm.

**Comment by dianaxxyyzz** [comment]
Yes , it is very strange ,I have Nexus 5 with Maru Os  , who run debian.rootfs.tar.gz using lxc .

It works good , no problem , even selinux is enforce .

I ported Maru Os for Lg G4 phone , and I'm using same debian.rootfs.tar.gz like in Nexus 5 .

I have even selinux on my android G4 phone , permissive . Nexus 5 have it enforced.

I have same version an same configuration lxc like in nexus 5 ,of course.

I do not unedstand why for nexus 5 works to log in but for my Lg g4 do not works , I'm using same debian.rootfs.tar.gz .

**Comment by stgraber** [comment]
So could be that the system error is coming from some missing kernel features then, as I doubt both your phones are using the exact same kernel.

I'd suggest you strace that passwd process and see exactly what's failing with that system error, that may point you to a missing syscall or other kernel config.

**Comment by brauner** [comment]
One thing to keep in mind is that last time I used `Debian Jessie` containers they appear to be functional when they are really not: the init system is not correctly initialized and the system is in a very degraded state. So it's not immediately obvious that something is wrong.

**Comment by dianaxxyyzz** [comment]
They use a similar kernel but not identical , also I added all kernel features I found at compile time ,inspired from Nexus 5 kernel . It may be kernel ,I always hoped is just missing something from kernel configuration that I can select at compile time.

I thinked it may be beucause android selinux , but mine is permisive , on nexus 5 was enforcing and everything worked. 
It is strange that same lxc configuration and same debian.rootfs.tgz give this error .

I try to see what is the correct way to strace that passwd process.

**Comment by dianaxxyyzz** [comment]
[output.txt](https://github.com/lxc/lxc/files/627828/output.txt)

Here is the strace of passwd , when I try to change password.

**Comment by dianaxxyyzz** [comment]
i see this : stat64("/dev/pts/8", 0xff983aa8)        = -1 ENOENT (No such file or directory) 
Maybe this is the problem? Who knows....

**Comment by pdsouza** [comment]
Thanks for the input guys!

To close the loop for anyone else affected by this: I ended up tracing this to a kernel audit issue for PAM authentication in secondary namespaces that affects kernels < v3.15.

The fix is a backported 3.15 kernel patch, it's available here with some more info on this issue: https://github.com/maruos/maruos/wiki/Porting-Guide#container-login-failures

I have verified with @dianaxxyyzz that this fixes his login issue on the LG G4 as well.
